### ISSUE TITLE
NullPointerException during notification

### ISSUE THREAD

**Issue by sephiroth-j** [DependencyTrack/dependency-track#818]
### Current Behavior:
NullPointerException  after bom upload with configured e-mail notification of level INFORMATIONAL, scope PORTFOLIO, groups NEW_VULNERABLE_DEPENDENCY, BOM_PROCESSED and limited to single project.

### Steps to Reproduce:
- have a project in Dependency-Track 
- create a notification limited to this project
- upload new bom for this project (e.g. using Jenkins plugin)

### Expected Behavior:
no exception and e-mail should have been sent

### Environment:
  
  - Dependency-Track Version: 3.8
  - Distribution: Docker
  - BOM Format & Version: CycloneDX v1.2 xml
  - Database Server: PostgreSQL 
  - Browser: 

### Additional Details:
**notification config**
```json
{
	"name": "Portfolio Alert",
	"enabled": true,
	"scope": "PORTFOLIO",
	"notificationLevel": "INFORMATIONAL",
	"projects": [
		{
			"name": "XXXX",
			"version": "SNAPSHOT",
			"uuid": "fdd8efad-24cb-4fe7-84b2-7c477d17a81f",
			"lastBomImport": 1604059084811,
			"lastBomImportFormat": "CycloneDX null",
			"lastInheritedRiskScore": 0.0,
			"active": true
		}
	],
	"notifyOn": [
		"NEW_VULNERABLE_DEPENDENCY",
		"BOM_PROCESSED"
	],
	"publisher": {
		"name": "Email",
		"description": "Sends notifications to an email address",
		"publisherClass": "org.dependencytrack.notification.publisher.SendMailPublisher",
		"templateMimeType": "text/plain",
		"defaultPublisher": true,
		"uuid": "0fcf2408-79a6-413c-a02a-e53288c01875"
	},
	"publisherConfig": "{\"destination\":\"XXX@YYY.ZZZ\"}",
	"uuid": "43020723-8635-4f02-91e1-1a4cb385df89"
}

```

**stacktrace**
```
11:58:06.672 ERROR [LoggableUncaughtExceptionHandler] An unknown error occurred in an asynchronous event or notification thread
java.lang.NullPointerException: Value in JsonObjects name/value pair cannot be null
        at org.glassfish.json.JsonObjectBuilderImpl.validateValue(JsonObjectBuilderImpl.java:222)
        at org.glassfish.json.JsonObjectBuilderImpl.add(JsonObjectBuilderImpl.java:90)
        at org.dependencytrack.util.NotificationUtil.toJson(NotificationUtil.java:303)
        at org.dependencytrack.notification.publisher.Publisher.prepareTemplate(Publisher.java:84)
        at org.dependencytrack.notification.publisher.SendMailPublisher.inform(SendMailPublisher.java:53)
        at org.dependencytrack.notification.NotificationRouter.inform(NotificationRouter.java:63)
        at alpine.notification.NotificationService.lambda$alertSubscriber$0(NotificationService.java:100)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by msymons** [comment]
@sephiroth-j , this is a known issue with Dependency-Track 3.8.0 notifications when using schemaVersion 1.2 BOM.  It is fixed in the upcoming DT v4.0.0

The work-around is to configure your BOM generation to use schemaVersion 1.1.  And then remember to revert that configuration when you upgrade to DT 4.0...  schemaVersion 1.2 does give us some nice goodies.

**Comment by sephiroth-j** [comment]
@msymons thank you for the information.

**Comment by stevespringett** [comment]
Closing. Fixed in v4.0 - to be released soon.
